versions/development/SPEC.md

```diff
@@ -698,6 +699,48 @@ Pair values can also be specified within the [workflow inputs JSON](https://gith
 }
 ```
 
+### Optional literals
+
+Any non-optional value can be wrapped into an optional value by assigning it to an optional variable:
+
+```WDL
+Int? maybe_integer = 5
+```
+
+The `None` literal is the value that an optional has when it is not defined.
+
+```WDL
+Int? maybe_integer = None
+# maybe_integer is not defined
+Int? maybe_numeral = 5
+Int certain_integer = 3
+# maybe_numeral is a defined optional
+is_defined(maybe_integer) # Evaluates to false
+is_defined(maybe_numeral) # Evaluates to true
+maybe_integer == None # Evaluates to true, same as !is_defined(maybe_integer)
+maybe_integer != None # Evaluates to false, same as is_defined(maybe_integer)
+certain_integer == None # This will cause an error. Since None only makes sense when dealing with optionals.
+```
+
+This is useful in a number of cases, for example when an output of a command depends on a certain flag:
+```wdl
+task test
+  input {
+    File inputFile
+    String outputPath
+    Boolean log
+  }
+  command {
+    test ~{true="--log" false="" log} inputFile outputPath
+  }
+  output {
+    File? logFile = if log then "test.log" else None
```

Comment:
is this actually needed? what happens now if you just have 
```wdl
  File? logFile = "test.log"
```

Reply:
Then `logFile` will always be `test.log` even if there is no `test.log` created.